int32 id

---

bool success
AgentMsg agent
